numba/tests/test_dufunc.py

```diff
@@ -0,0 +1,28 @@
+from __future__ import print_function, absolute_import, division
+from numba import unittest_support as unittest
+
+import numpy as np
+
+from numba.npyufunc import dufunc
+
+def pyuadd(a0, a1):
+    return a0 + a1
+
+class TestDUFunc(unittest.TestCase):
+
+    def test_frozen(self):
+        duadd = dufunc.DUFunc(pyuadd, nopython=True)
+        self.assertFalse(duadd.frozen)
+        duadd.frozen = True
```

Comment:
Having a "frozen" flag that can be set to `True` and then not back to `False` seems a little odd - would it not be cleaner to have a `freeze()` method instead?